sklearn/cluster/_kmeans.py

```diff
@@ -863,6 +864,38 @@ def _check_test_data(self, X):
 
         return X
 
+    def _check_mkl_vcomp(self, X, n_samples):
+        """Warns when vcomp and mkl are both present"""
+        # The BLAS call inside a prange in lloyd_iter_chunked_dense is known to
+        # cause a small memory leak when there are less chunks than the number
+        # of available threads. It only happens when the OpenMP library is
+        # vcomp (microsoft OpenMP) and the BLAS library is MKL. see #18653
+        if sp.issparse(X):
+            return
+
+        active_threads = int(np.ceil(n_samples / 256))
```

Comment:
This `256` should be defined in a module level constant (e.g. `PER_THREAD_CHUNK_SIZE` or just `CHUNK_SIZE`).